### PR TITLE
fixing bug sa-125 and 124

### PR DESCRIPTION
**Pull request by ar-india**
resolution has been explained in the related bug

### CODE REVIEW

File: ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_snapshot_collection.py [hunk 1]
```diff
@@ -177,18 +178,19 @@
...
     if utils.is_null_or_empty(snapcoll_name):
         return (False, False, "Create snapshot collection failed. snapshot collection name is not present.", {})
     try:
-        snapcoll_resp = client_obj.snapshot_collections.get(id=None, name=snapcoll_name)
+        snapcoll_resp = client_obj.snapshot_collections.get(id=None, name=snapcoll_name, volcoll_name=volcoll_name)
         if utils.is_null_or_empty(snapcoll_resp):
             params = utils.remove_null_args(**kwargs)
             snapcoll_resp = client_obj.snapshot_collections.create(name=snapcoll_name, **params)
-            return (True, True, f"Created snapshot collection '{snapcoll_name}' successfully.", {})
+            return (True, True, f"Created snapshot collection '{snapcoll_name}' inside volume collection '{volcoll_name}' successfully.", {})
```
Comment (suneeth51) [inline ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_snapshot_collection.py]: nit: Created snapshot collection '{snapcoll_name}' for volume collection '{volcoll_name}' successfully

File: ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_snapshot_collection.py [hunk 2]
```diff
@@ -177,18 +178,19 @@
...
         else:
-            return (False, False, f"Snapshot collection '{snapcoll_name}' cannot be created as it is already present.", {})
+            return (False, False, f"Snapshot collection '{snapcoll_name}' inside volume collection '{volcoll_name}' cannot be created as it is already present.", {})
```
Comment (suneeth51) [inline ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_snapshot_collection.py]: nit: Snapshot collection '{snapcoll_name}' for volume collection '{volcoll_name}' cannot be created as it is already present.

File: ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_snapshot_collection.py [hunk 3]
```diff
@@ -212,18 +214,18 @@ def update_snapcoll(
...
 
     if utils.is_null_or_empty(snapcoll_name):
         return (False, False, "Snapshot collection deletion failed as snapshot collection name is not present.", {})
 
     try:
-        snapcoll_resp = client_obj.snapshot_collections.get(id=None, name=snapcoll_name)
+        snapcoll_resp = client_obj.snapshot_collections.get(id=None, name=snapcoll_name, volcoll_name=volcoll_name)
         if utils.is_null_or_empty(snapcoll_resp):
-            return (False, False, f"Snapshot collection '{snapcoll_name}' not present to delete.", {})
+            return (False, False, f"Snapshot collection '{snapcoll_name}' inside volume collection '{volcoll_name}' not present to delete.", {})
```
Comment (suneeth51) [inline ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_snapshot_collection.py]: nit: Snapshot collection '{snapcoll_name}' for volume collection '{volcoll_name}' not present to delete.

File: ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_snapshot_collection.py [hunk 4]
```diff
@@ -212,18 +214,18 @@ def update_snapcoll(
...
         else:
             client_obj.snapshot_collections.delete(id=snapcoll_resp.attrs.get("id"))
-            return (True, True, f"Snapshot collection '{snapcoll_name}' deleted successfully.", {})
+            return (True, True, f"Snapshot collection '{snapcoll_name}' inside volume collection '{volcoll_name}' deleted successfully.", {})
```
Comment (suneeth51) [inline ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_snapshot_collection.py]: nit: Snapshot collection '{snapcoll_name}' for volume collection '{volcoll_name}' deleted successfully

**Review by suneeth51 — APPROVED** [review]
lgtm with few minor comments.

File: ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_snapshot_collection.py (hunk 2 above)
Comment (ar-india) [reply ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_snapshot_collection.py]: k

**Pull request merged**